### ISSUE TITLE
False positive `consider-using-enumerate` when using augmented assignment

### ISSUE THREAD

**Issue by macdjord** [PyCQA/pylint#11304]
### Bug description
Code such as this generates a `consider-using-enumerate` warning:
```python
for i in range(len(self._values)):
    self._values[i] *= normalization_factor
```
It is not possible to usefully convert this code to use `enumerate()`, since we never explicitly read from the values of the sequence we are iterating over, only write to them.

### Configuration

```ini

```

### Command used

```shell
N/A
```

### Pylint output

```python
N/A
```

### Expected behavior

No such warning is emitted.

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
```

### OS / Environment

Windows 11

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
We right implicitely during the *=, what's wrong with this: ?

```python
 for i, value in enumerate(self._values):
      self._values[i] = value * normalization_factor
```
